s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/RecordProcessor.java

```diff
@@ -53,44 +53,39 @@ public static List<SourceRecord> processRecords(final Iterator<List<AivenS3Sourc
         return results;
     }
 
-    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
-    static List<SourceRecord> createSourceRecords(final List<AivenS3SourceRecord> aivenS3SourceRecordList,
+    static SourceRecord createSourceRecord(final AivenS3SourceRecord aivenS3SourceRecord,
             final S3SourceConfig s3SourceConfig, final Optional<Converter> keyConverter, final Converter valueConverter,
             final Map<String, String> conversionConfig, final OutputWriter outputWriter,
             final Set<String> failedObjectKeys, final OffsetManager offsetManager) {
 
-        final List<SourceRecord> sourceRecordList = new ArrayList<>();
-        for (final AivenS3SourceRecord aivenS3SourceRecord : aivenS3SourceRecordList) {
-            LOGGER.info(" ******* CSR key ******** {}", aivenS3SourceRecord.getObjectKey());
-            final String topic = aivenS3SourceRecord.getToTopic();
-            final Optional<SchemaAndValue> keyData = keyConverter
-                    .map(c -> c.toConnectData(topic, aivenS3SourceRecord.key()));
-
-            outputWriter.configureValueConverter(conversionConfig, s3SourceConfig);
-            valueConverter.configure(conversionConfig, false);
-            try {
-                final SchemaAndValue schemaAndValue = valueConverter.toConnectData(topic, aivenS3SourceRecord.value());
-                offsetManager.updateCurrentOffsets(aivenS3SourceRecord.getPartitionMap(),
-                        aivenS3SourceRecord.getOffsetMap());
-                aivenS3SourceRecord.setOffsetMap(offsetManager.getOffsets().get(aivenS3SourceRecord.getPartitionMap()));
-                sourceRecordList.add(aivenS3SourceRecord.getSourceRecord(topic, keyData, schemaAndValue));
-            } catch (DataException e) {
-                LOGGER.error("Error in reading s3 object stream " + e.getMessage());
-                failedObjectKeys.add(aivenS3SourceRecord.getObjectKey());
-                throw e;
-            }
+        final String topic = aivenS3SourceRecord.getToTopic();
+        final Optional<SchemaAndValue> keyData = keyConverter
+                .map(c -> c.toConnectData(topic, aivenS3SourceRecord.key()));
+
+        outputWriter.configureValueConverter(conversionConfig, s3SourceConfig);
+        valueConverter.configure(conversionConfig, false);
+        try {
+            final SchemaAndValue schemaAndValue = valueConverter.toConnectData(topic, aivenS3SourceRecord.value());
+            offsetManager.updateCurrentOffsets(aivenS3SourceRecord.getPartitionMap(),
+                    aivenS3SourceRecord.getOffsetMap());
+            aivenS3SourceRecord.setOffsetMap(offsetManager.getOffsets().get(aivenS3SourceRecord.getPartitionMap()));
+            return aivenS3SourceRecord.getSourceRecord(topic, keyData, schemaAndValue);
+        } catch (DataException e) {
+            LOGGER.error("Error in reading s3 object stream " + e.getMessage());
```

Comment:
Log statements should be 
LOGGER.error("Error in reading s3 object stream {}", e.getMessage());
or
LOGGER.error("Error in reading s3 object stream",e);